src/Item/Item.service.ts

```diff
@@ -0,0 +1,24 @@
+import { CollectionService } from '../Collection/Collection.service'
+import { Item } from './Item.model'
+import { ItemAttributes } from './Item.types'
+
+export class ItemService {
+  private collectionService = new CollectionService()
+
+  public async isOwnedOrManagedBy(
+    id: string,
+    ethAddress: string
+  ): Promise<boolean> {
+    const dbItem = await Item.findOne<ItemAttributes>(id)
+    if (dbItem?.urn_suffix && dbItem?.collection_id) {
```

Comment:
Nitpick: We could avoid `?.` if you want:

```ts
if (!dbItem) {
  return false
} else if (dbItem.urn_suffix && dbItem.collection_id) {
  return this.collectionService.isOwnedOrManagedBy(
    dbItem.collection_id,
    ethAddress
  )
} else {
  return dbItem.eth_address === dbItem.eth_address
}
```

not sure if more legible, just throwing it out there

Reply:
It seems pretty much the same to me, but I changed it anyways.